src/main/java/it/smartcommunitylab/aac/webauthn/persistence/WebAuthnYubicoCredentialsRepository.java

```diff
@@ -0,0 +1,103 @@
+        this.webAuthnCredentialsRepository = webAuthnCredentialsRepository;
+        this.providerId = provider;
+    }
+
+    @Override
+    public Set<PublicKeyCredentialDescriptor> getCredentialIdsForUsername(String username) {
+        try {
+            WebAuthnUserAccount account = userAccountRepository.findByProviderAndUsername(providerId, username);
+            final Set<WebAuthnCredential> credentials = account.getCredentials();
+            Set<PublicKeyCredentialDescriptor> descriptors = new HashSet<PublicKeyCredentialDescriptor>();
+            for (WebAuthnCredential c : credentials) {
+                PublicKeyCredentialDescriptor descriptor = PublicKeyCredentialDescriptor.builder()
+                        .id(c.getCredentialId()).type(PublicKeyCredentialType.PUBLIC_KEY).transports(c.getTransports())
+                        .build();
+                descriptors.add(descriptor);
+            }
+            return descriptors;
+        } catch (Exception e) {
+        }
+        return new HashSet<>();
+    }
+
+    @Override
+    public Optional<ByteArray> getUserHandleForUsername(String username) {
+        WebAuthnUserAccount account = userAccountRepository.findByProviderAndUsername(providerId, username);
+        if (account != null) {
+            return Optional.of(account.getUserHandle());
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<String> getUsernameForUserHandle(ByteArray userHandle) {
+        try {
+            WebAuthnUserAccount account = userAccountRepository.findByUserHandle(userHandle.getBase64());
+            if (account == null) {
+                return Optional.empty();
+            }
+            return Optional.of(account.getUsername());
+        } catch (Exception e) {
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<RegisteredCredential> lookup(ByteArray credentialId, ByteArray userHandle) {
+        WebAuthnUserAccount acc = userAccountRepository.findByUserHandle(userHandle.getBase64());
+        if (acc == null) {
+            return Optional.empty();
+        }
+        Set<WebAuthnCredential> credentials = acc.getCredentials();
+        for (final WebAuthnCredential cred : credentials) {
+            if (cred.getCredentialId().equals(credentialId)) {
+                return Optional.of(cred.getRegisteredCredential());
+            }
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Set<RegisteredCredential> lookupAll(ByteArray credentialId) {
+        // In our database the credentialID already has a unique constraint
+        Set<RegisteredCredential> s = new HashSet<>();
+        WebAuthnCredential cred = webAuthnCredentialsRepository.findByCredentialId(credentialId.getBase64());
+        if (cred != null) {
+            s.add(cred.getRegisteredCredential());
+        }
+        return s;
+    }
+
+    public WebAuthnUserAccount save(WebAuthnUserAccount infos) {
```

Comment:
why this method here? credentials repository should handle credentials, not add user accounts

Reply:
This method is not needed anymore, removing it